### ISSUE TITLE
Error when formatting document or selection with only PLS extension enabled in VSCode

### ISSUE THREAD

**Issue by dklawren** [FractalBoy/perl-language-server#50]
VSCode for Windows 1.57.1
WSL2 with Fedora 34 connecting with Remote: WSL in VSCode

```bash
❯ perltidy -v
This is perltidy, v20210625 

Copyright 2000-2021, Steve Hancock

Perltidy is free software and may be copied under the terms of the GNU
General Public License, which is included in the distribution files.

Complete documentation for perltidy can be found using 'man perltidy'
or on the internet at http://perltidy.sourceforge.net.
```

```json
{
    "[perl]": {
        "editor.defaultFormatter": "FractalBoy.pls"
    },
    "perl.syntax.perl": "perl",
    "perl.perlcritic.perlcriticrc": "~/.perlcriticrc",
    "perl.perltidyrc": "~/.perltidyrc",
    "perl.inc": [
        "/home/dkl/perl5"
    ]
}
```

I am getting the error in PLS output tab when using Control-Shift-P -> Format Document or Format Selection

```bash
[Error - 2:33:01 PM] Request textDocument/formatting failed.
  Message: Perltidy failed to format the text.
  Code: -32700 
You may not specify a destination array and -st together
 (-pbp contains -st; see manual)
```

Additional Log output:

```
[2021-07-06 14:39:47.122] [exthost] [info] ExtensionService#_doActivateExtension FractalBoy.pls {"startup":false,"extensionId":{"value":"FractalBoy.pls","_lower":"fractalboy.pls"},"activationEvent":"onLanguage:perl"}
...
[2021-07-06 14:39:54.544] [exthost] [error] [FractalBoy.pls] provider FAILED
[2021-07-06 14:39:54.544] [exthost] [error] Error: Perltidy failed to format the text.
	at /home/dkl/.vscode-server/extensions/fractalboy.pls-0.0.9/out/extension.js:1:30349
	at /home/dkl/.vscode-server/extensions/fractalboy.pls-0.0.9/out/extension.js:1:30643
	at Immediate.<anonymous> (/home/dkl/.vscode-server/extensions/fractalboy.pls-0.0.9/out/extension.js:1:31006)
	at processImmediate (internal/timers.js:461:21)
[2021-07-06 14:40:01.209] [exthost] [warning] TextEditor is closed/disposed
```

Any idea what could be happening? I acan use the perltidy extension along side this one but I would love to be able to just have the one Perl extension to handle what I need. 

Let me know if I can give more details.

**Comment by FractalBoy** [comment]
Can you try fixing your .perltidyrc?

**Comment by FractalBoy** [comment]
Nevermind - I'm guessing you have only `-pbp` in your .perltidyrc? I think the issue is that PLS appends `-se` to the args, but `-pbp` already implies `-se`.

The error message is a bit misleading - `-st` is not added by PLS.

**Comment by dklawren** [comment]
> 
> 
> Can you try fixing your .perltidyrc?

Hmm, If I use the same .perltidyrc using the sfodje.perltidy extension it works fine. Also if I run perltidy on a Perl file from the command line in the same directory as the .perltidyrc it also works without complaint.

> Nevermind - I'm guessing you have only `-pbp` in your .perltidyrc? I think the issue is that PLS appends `-se` to the args, but `-pbp` already implies `-se`.
> 
> The error message is a bit misleading - `-st` is not added by PLS.

Here is the full .perltidyrc

```
-pbp     # Start with Perl Best Practices
-w       # Show all warnings
-iob     # Ignore old breakpoints
-l=80    # 80 characters per line
-vmll
-ibc
-iscl
-hsc
-mbl=2   # No more than 2 blank lines
-i=2     # Indentation is 2 columns
-ci=2    # Continuation indentation is 2 columns
-vt=0    # Less vertical tightness
-pt=2    # High parenthesis tightness
-bt=2    # High brace tightness
-sbt=2   # High square bracket tightness
-wn      # Weld nested containers
-isbc    # Don't indent comments without leading space
```

**Comment by FractalBoy** [comment]
PLS uses the Perl::Tidy::perltidy API directly, so I suppose it functions a little differently than the command line utility.

There is this caveat in the [Perl::Tidy documentation](https://metacpan.org/pod/Perl::Tidy#NOTES-ON-FORMATTING-PARAMETERS):

>If the -pbp style is used it will typically be necessary to also specify a -nst flag. This is necessary to turn off the -st flag contained in the -pbp parameter set which otherwise would direct the output stream to the standard output.

**Comment by FractalBoy** [comment]
I'm able to format a file with this perltidyrc:

```
-pbp
-nst
```

For some reason, I needed to restart PLS in order to pick up the new config - I will investigate that.

Can you try adding -nst to your config and reloading?

**Comment by dklawren** [comment]
Thanks. Added -nst did in fact fix the issue for me.

```
-pbp     # Start with Perl Best Practices
-w       # Show all warnings
-iob     # Ignore old breakpoints
-l=80    # 80 characters per line
-vmll
-ibc
-iscl
-hsc
-mbl=2   # No more than 2 blank lines
-i=2     # Indentation is 2 columns
-ci=2    # Continuation indentation is 2 columns
-vt=0    # Less vertical tightness
-pt=2    # High parenthesis tightness
-bt=2    # High brace tightness
-sbt=2   # High square bracket tightness
-wn      # Weld nested containers
-isbc    # Don't indent comments without leading space
-nst     # Necessary to turn off the -st flag contained in the -pbp parameter
```

I also noticed that I did need to restart VSCode to pick up the new change but I normally do that out of habit anyway. Will leave that to another issue.

Thanks!

**Comment by FractalBoy** [comment]
The issue where a restart was required will be fixed in 0.85 which I just uploaded.
